nautilus_trader/analysis/reports.pyx

```diff
@@ -80,14 +75,22 @@ cdef class ReportProvider:
         if not orders:
             return pd.DataFrame()
 
-        cdef list filled_orders = [
-            self._order_to_dict(o) for o in orders if o.state == OrderState.FILLED
-        ]
-
+        cdef list filled_orders = [o.to_dict() for o in orders if o.state == OrderState.FILLED]
         if not filled_orders:
             return pd.DataFrame()
 
-        return pd.DataFrame(data=filled_orders).set_index("client_order_id").sort_index()
+        report = pd.DataFrame(data=filled_orders).set_index("client_order_id").sort_index()
+        report["timestamp_ns"] = [nanos_to_unix_dt(row) for row in report["timestamp_ns"]]
+        report["ts_filled_ns"] = [nanos_to_unix_dt(row) for row in report["ts_filled_ns"]]
```

Comment:
```suggestion
        report["timestamp_ns"] = report["timestamp_ns"].apply(lambda row: nanos_to_unix_dt(row))
        report["ts_filled_ns"] = report["ts_filled_ns"].apply(lambda row: nanos_to_unix_dt(row))
```

Applying it should also work and much faster. Probably don't need to use lambda either :p

Reply:
Good point! I'll try out this suggestion tomorrow, I'm always looking for refinements and simplifications 😄